libicu 75+ requires cpp+17

A word of warning more then a bug?

In case anyone else encounters compilation issues similar to:

```
g++ -std=c++14 -O2 -ggdb -Iinclude -Istfl -Ifilter -I. -Irss -I/home/carno/dokumenty/github/newsboat/target/cxxbridge/ -Werror -Wall -Wextra -Wunreachable-code -DLOCALEDIR='"/home/carno/.local/share/locale"' -DCATCH_AMALGAMATED_CUSTOM_MAIN -I/usr/include/x86_64-linux-gnu  -I/usr/include/libxml2  -I/usr/include/json-c  -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600   -MD -MP -MF .deps/src/colormanager.cpp -o src/colormanager.o -c src/colormanager.cpp
In file included from /usr/include/unicode/uenum.h:25,
                 from /usr/include/unicode/ucnv.h:52,
                 from /usr/include/libxml2/libxml/encoding.h:31,
                 from /usr/include/libxml2/libxml/parser.h:812,
                 from include/utils.h:6,
                 from src/colormanager.cpp:8:
/usr/include/unicode/localpointer.h:561:26: error: ‘auto’ parameter not permitted in this context
  561 | template <typename Type, auto closeFunction>
      |                          ^~~~
/usr/include/unicode/localpointer.h:573:76: error: template argument 2 is invalid
  573 |     explicit LocalOpenPointer(std::unique_ptr<Type, decltype(closeFunction)> &&p)
      |                                                                            ^
/usr/include/unicode/localpointer.h:583:78: error: template argument 2 is invalid
  583 |     LocalOpenPointer &operator=(std::unique_ptr<Type, decltype(closeFunction)> &&p) {
      |                                                                              ^
/usr/include/unicode/localpointer.h:599:59: error: template argument 2 is invalid
  599 |     operator std::unique_ptr<Type, decltype(closeFunction)> () && {
      |                                                           ^
/usr/include/unicode/uenum.h:69:1: note: invalid template non-type parameter
   69 | U_DEFINE_LOCAL_OPEN_POINTER(LocalUEnumerationPointer, UEnumeration, uenum_close);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/unicode/ucnv.h:597:1: note: invalid template non-type parameter
  597 | U_DEFINE_LOCAL_OPEN_POINTER(LocalUConverterPointer, UConverter, ucnv_close);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
make: *** [Makefile:193: src/colormanager.o] Error 1
```

it most likely is due to your distro bumping `libicu` to one of the recent version (I just encountered it with Debian Sid after they jumped to 76.

I didn't dig to understand if `newsboat` depends on `libicu` or is it just indirect dependency via `libxml2` (or something else?).

Compilation works with

```diff
diff --git i/Makefile w/Makefile
index 3e3db046..c8d661f5 100644
--- i/Makefile
+++ w/Makefile
@@ -25,7 +25,7 @@ DEFINES=-DLOCALEDIR='"$(localedir)"' -DCATCH_AMALGAMATED_CUSTOM_MAIN
 WARNFLAGS=-Werror -Wall -Wextra -Wunreachable-code
 INCLUDES=-Iinclude -Istfl -Ifilter -I. -Irss -I$(CARGO_TARGET_DIR)/cxxbridge/
 # Keep in sync with c++ version specified in FFI build.rs
-BARE_CXXFLAGS=-std=c++14 -O2 -ggdb $(INCLUDES)
+BARE_CXXFLAGS=-std=c++17 -O2 -ggdb $(INCLUDES)
 LDFLAGS+=-L.
 
 # Constants
```

Refs:
* https://github.com/unicode-org/icu/releases/tag/release-75-rc
* https://github.com/ledger/ledger/issues/2375#issuecomment-2287384274

--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -237,14 +237,6 @@ task:
                 cxx_package: g++-14
                 cc: gcc-14
                 cxx: g++-14
-        - name: Rust stable, Clang 4.0 (Ubuntu 18.04)
-          container:
-            greedy: true
-            dockerfile: docker/ubuntu_18.04-build-tools.dockerfile
-            docker_arguments:
-                cxx_package: clang-4.0
-                cc: clang-4.0
-                cxx: clang++-4.0
         - name: Rust stable, Clang 5.0 (Ubuntu 18.04)
           container:
             greedy: true

--- a/Makefile
+++ b/Makefile
@@ -25,7 +25,7 @@ DEFINES=-DLOCALEDIR='"$(localedir)"' -DCATCH_AMALGAMATED_CUSTOM_MAIN
 WARNFLAGS=-Werror -Wall -Wextra -Wunreachable-code
 INCLUDES=-Iinclude -Istfl -Ifilter -I. -Irss -I$(CARGO_TARGET_DIR)/cxxbridge/
 # Keep in sync with c++ version specified in FFI build.rs
-BARE_CXXFLAGS=-std=c++14 -O2 -ggdb $(INCLUDES)
+BARE_CXXFLAGS=-std=c++17 -O2 -ggdb $(INCLUDES)
 LDFLAGS+=-L.
 
 # Constants

--- a/README.md
+++ b/README.md
@@ -64,7 +64,10 @@ Newsboat can be compiled.
     UPDATE doc/newsboat.asciidoc IF YOU CHANGE THIS LIST
 -->
 - GNU Make 4.0 or newer
-- GCC 5.0 or newer, or Clang 3.6 or newer
+- C++ compiler that accepts `-std=c++17` option: GCC 5.0 or newer, or Clang 5 or
+  newer. If you have libicu 76 or newer, than C++17 support is required as
+  Newsboat indirectly depends on libicu headers via libxml2. Newsboat 2.40 (ETA
+  2025.06.22) is expected to require a compiler that _supports_ C++17
 - Stable [Rust](https://www.rust-lang.org/en-US/) and Cargo (Rust's package
     manager) (1.81.0 or newer; might work with older versions, but we don't
     check that)

--- a/doc/newsboat.asciidoc
+++ b/doc/newsboat.asciidoc
@@ -69,7 +69,11 @@ Debian and derivatives, headers are in `-dev` packages, e.g. `libsqlite3-dev`.)
 
 // UPDATE README.md IF YOU CHANGE THIS LIST
 - GNU Make 4.0 or newer
-- GCC 5.0 or newer, or Clang 3.6 or newer
+- C{plus}{plus} compiler that accepts `-std=c{plus}{plus}17` option: GCC 5.0 or
+  newer, or Clang 5 or newer. If you have libicu 76 or newer, than
+  C{plus}{plus}17 support is required as Newsboat indirectly depends on libicu
+  headers via libxml2. Newsboat 2.40 (ETA 2025.06.22) is expected to require
+  a compiler that _supports_ C{plus}{plus}17
 - Stable https://www.rust-lang.org/en-US/[Rust] and Cargo (Rust's package
   manager) (1.81.0 or newer; might work with older versions, but we don't check
   that)

--- a/rust/libnewsboat-ffi/build.rs
+++ b/rust/libnewsboat-ffi/build.rs
@@ -1,7 +1,7 @@
 fn add_cxxbridge(module: &str) {
     cxx_build::bridge(format!("src/{module}.rs"))
         // Keep in sync with c++ version specified in Makefile
-        .std("c++14")
+        .std("c++17")
         // Disable warnings in generated code, since we can't affect them directly. We have to do
         // this because these warnings are turned into errors when we pass `-D warnings` to Cargo
         // on CI.

